lightningd/gossip_control.c

```diff
@@ -71,13 +71,13 @@ static void get_txout(struct subd *gossip, const u8 *msg)
 
 	op = wallet_outpoint_for_scid(gossip->ld->wallet, scid, scid);
 
-	if (op) {
+	if (op && !op->spendheight) {
```

Comment:
We could also just push that down to the sqlite query by adding an argument `includespent`. That'd allow us to skip materializing the result in memory just to drop it again immediately.

Reply:
Also `op` is allocated off of `scid` which could be a long-lived object (it currently isn't).